File: account-service/src/main/java/com/food/delivery/accountservice/company/CompanyClient.java
package com.food.delivery.accountservice.company;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import reactivefeign.spring.config.ReactiveFeignClient;
import reactor.core.publisher.Mono;

@ReactiveFeignClient(value = "company-service", fallbackFactory = CompanyClientFallbackFactory.class)
public interface CompanyClient {

	@GetMapping("/company/{companyId}")
	Mono<CompanyRest> getCompany(@PathVariable String companyId);

}

File: menu-service/src/main/kotlin/com/food/delivery/menuservice/company/CompanyClient.kt
package com.food.delivery.menuservice.company

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import reactivefeign.spring.config.ReactiveFeignClient
import reactor.core.publisher.Mono

@ReactiveFeignClient(value = "company-service", fallbackFactory = CompanyClientFallbackFactory::class)
interface CompanyClient {

	@GetMapping("/company/{companyId}")
	fun getCompany(@PathVariable companyId: String): Mono<CompanyRest>
}
